doc/ipython-notebooks/gaussian_process/variational_classifier.ipynb

```diff
@@ -629,14 +733,380 @@
+     "collapsed": false,
+     "input": [
+      "def extract1(inf):\n",
+      "    \"\"\"\n",
+      "    This function is used to process raw data of the banana data set\n",
+      "        \n",
+      "    Parameters:\n",
+      "        inf - the path of the raw data\n",
+      "    Returns:\n",
+      "        x_train, x_test - features for training and testing\n",
+      "        y_train, y_test - labels for training and testing\n",
+      "        x1s, x2s  -for ploting training or testing points\n",
+      "    \"\"\"\n",
+      "\n",
+      "    random.seed(1)\n",
+      "    x=[[], []]\n",
+      "    y=[[], []]\n",
+      "    x1s=[{}, {}]\n",
+      "    x2s=[{}, {}]\n",
+      "\n",
+      "    for line in open(inf):\n",
+      "        line=line.strip()\n",
+      "        if line.startswith(\"@\"):\n",
+      "            continue\n",
+      "        choice=random.randint(0,1)\n",
+      "        if choice==0:\n",
+      "            #testing set\n",
+      "            idx=1\n",
+      "        else:\n",
+      "            #training set\n",
+      "            idx=0\n",
+      "        info=map(float,line.split(','))\n",
+      "        x[idx].append(info[:-1])\n",
+      "        y[idx].append(info[-1])\n",
+      "        if info[-1]>0:\n",
+      "            #positive cases\n",
+      "            container_x1=x1s[idx].setdefault(1,[])\n",
+      "            container_x2=x2s[idx].setdefault(1,[])\n",
+      "        else:\n",
+      "            #negative cases\n",
+      "            container_x1=x1s[idx].setdefault(0,[])\n",
+      "            container_x2=x2s[idx].setdefault(0,[])\n",
+      "        container_x1.append(info[0])\n",
+      "        container_x2.append(info[1])\n",
+      "\n",
+      "    x_train = np.asarray(x[0]).T\n",
+      "    y_train = np.asarray(y[0])\n",
+      "    x_test = np.asarray(x[1]).T\n",
+      "    y_test = np.asarray(y[1])\n",
+      "\n",
+      "    return x_train, y_train, x_test, y_test, x1s, x2s"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "from itertools import product\n",
+      "def plot(inf):\n",
+      "    \"\"\"\n",
+      "    This function is used to print the decision bounday and testing points\n",
+      "        \n",
+      "    Parameters:\n",
+      "        inf - the path of the raw data\n",
+      "    \"\"\"\n",
+      "    #obtain data points with labels from raw data\n",
+      "    (x_train, y_train, x_test, y_test, x1s, x2s)=extract1(inf)\n",
+      "    #binary classification problem (positive labels and negative labels)\n",
+      "    inference =SingleLaplacianInferenceMethod\n",
+      "    likelihood = LogitLikelihood()\n",
+      "    linesearch = 3\n",
+      "    #we show how parameters of GPC affect the decision boundary\n",
+      "    kernel_log_sigmas=[-5,0,1]#over-fit, right-fit, under-fit\n",
```

Comment:
Could you avoid doing comments in the same line as code? It makes thing look messy. Also add a couple of newlines here to decrease the density of characters